### PR TITLE
Add support for rendering Topics sections without topic group headings

### PR DESCRIPTION
**Pull request by dobromir-hristov**
Bug/issue #, if applicable: 100452122

## Summary

This PR refactors the TopicsSections to properly render even without a title.

image

## Dependencies

- https://github.com/apple/swift-docc/pull/393

## Testing

Use the attached DocC PR to generate an archive without titles for some of it's sections

Steps:
1. Assert sections are properly spaced without extra spacing
2. Assert there are no warnings in the console or regressions

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran `npm test`, and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

**Comment by dobromir-hristov** [comment]
@swift-ci test

File: src/components/DocumentationTopic/ContentTableSection.vue [hunk 1]
```diff
@@ -40,7 +41,8 @@ export default {
   props: {
     title: {
       type: String,
-      required: true,
+      default: '',
```
Comment (mportiz08) [inline src/components/DocumentationTopic/ContentTableSection.vue:44]: Why `''` as a default value?

Comment (mportiz08) [reply src/components/DocumentationTopic/ContentTableSection.vue:44]: I realize it still works with the conditional above, but it seems odd to me as opposed just making it optional without a default.

**Review by mportiz08 — APPROVED** [review]

**Comment by dobromir-hristov** [comment]
@swift-ci test

**Comment by dobromir-hristov** [comment]
@swift-ci test

**Pull request merged**